src/main/kotlin/com/openlattice/mechanic/upgrades/CreateStagingSchemaForExistingOrgs.kt

```diff
@@ -0,0 +1,107 @@
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.organizations.Organization
+import com.openlattice.postgres.DataTables
+import com.zaxxer.hikari.HikariDataSource
+import org.slf4j.LoggerFactory
+import java.util.*
+
+class CreateStagingSchemaForExistingOrgs(
+        private val toolbox: Toolbox,
+        private val assemblerConfiguration: AssemblerConfiguration
+) : Upgrade {
+
+    companion object {
+        private const val STAGING_SCHEMA = "staging"
+        private val logger = LoggerFactory.getLogger(CreateStagingSchemaForExistingOrgs::class.java)
+    }
+
+    override fun upgrade(): Boolean {
+        val allOrgs = HazelcastMap.ORGANIZATIONS.getMap(toolbox.hazelcast).values.toList()
+        val allPrincipalsToDbIds = getAllUserPrincipals()
+
+        allOrgs.forEach { createAndConfigureSchemaForOrg(it, allPrincipalsToDbIds) }
+        return true
+    }
+
+    override fun getSupportedVersion(): Long {
+        return Version.V2020_06_11.value
+    }
+
+    private fun getAllUserPrincipals(): Map<Principal, String> {
+        return HazelcastMap.PRINCIPALS.getMap(toolbox.hazelcast).values.associate {
+            it.principal to DataTables.quote(PostgresRoles.buildPostgresUsername(it))
+        }
+    }
+
+    private fun createAndConfigureSchemaForOrg(org: Organization, allPrincipals: Map<Principal, String>) {
+        logger.info("About to create staging schema for org ${org.title} [${org.id}]")
+
+        val members = org.members.mapNotNull { allPrincipals[it] } + DataTables.quote(PostgresRoles.buildOrganizationUserId(org.id))
+
+        connectToDatabase(org).let { hds ->
+            hds.connection.use { connection ->
+                connection.createStatement().use { stmt ->
+
+                    // create schema
+                    stmt.execute(CREATE_SCHEMA_SQL)
+
+                    // grant usage, create on schema to members (including org user)
+                    stmt.execute(grantPrivilegesSql(members.joinToString()))
+
+                    // update search path for users (including org user) to include staging schema
+                    members.forEach {
+                        stmt.execute(setSearchPathSql(
+                                it,
+                                true,
+                                AssemblerConnectionManager.MATERIALIZED_VIEWS_SCHEMA,
+                                STAGING_SCHEMA
+                        ))
+                    }
+
+                    // update search path for server user to include staging schema
+                    stmt.execute(setSearchPathSql(
+                            assemblerConfiguration.server["username"].toString(),
+                            false,
+                            AssemblerConnectionManager.INTEGRATIONS_SCHEMA,
+                            AssemblerConnectionManager.MATERIALIZED_VIEWS_SCHEMA,
+                            AssemblerConnectionManager.PUBLIC_SCHEMA,
```

Comment:
What do we have in the public schema?

Reply:
no idea -- this code roughly exists right now in AssemblerConnectionManager (minus the staging schema) on org creation so i just took this list from there. We do also revoke all permissions for user accounts on the public schema -- maybe this was somehow tied to old MV code?